### PR TITLE
[get-uri] Remove `fs-extra` dependency

### PR DESCRIPTION
**Pull request by alexparish**
Since the `get-uri` package requires Node.js 14 or higher, the `fs-extra` package can be removed and its features replaced with native Node.js functionality.

### CODE REVIEW

File: packages/get-uri/package.json [hunk 1]
```diff
@@ -51,8 +51,7 @@
   "dependencies": {
     "basic-ftp": "^5.0.2",
     "data-uri-to-buffer": "^6.0.2",
-    "debug": "^4.3.4",
-    "fs-extra": "^11.2.0"
```
Comment (ziebam) [inline packages/get-uri/package.json:55]: `@types/fs-extra` can also be removed from `devDependencies`.

File: packages/get-uri/src/file.ts [hunk 2]
```diff
@@ -42,11 +41,11 @@ export const file: GetUriProtocol<FileOptions> = async (
 
 		// `open()` first to get a file descriptor and ensure that the file
 		// exists.
-		const fd = await open(filepath, flags, mode);
+		const fd = await fsPromises.open(filepath, flags, mode);
 
 		// Now `fstat()` to check the `mtime` and store the stat object for
 		// the cache.
```
Comment (ziebam) [inline packages/get-uri/src/file.ts:47]: The comment is outdated here now.

File: packages/get-uri/package.json (hunk 1 above)
Comment (alexparish) [reply packages/get-uri/package.json:55]: Good spot, thanks!

File: packages/get-uri/src/file.ts (hunk 2 above)
Comment (alexparish) [reply packages/get-uri/src/file.ts]: Updated.

**Comment by alexparish** [comment]
Looking at the test failures now.

**Comment by alexparish** [comment]
Tests updated.

**Comment by alexparish** [comment]
Updated to work with Node 14.

File: packages/get-uri/src/file.ts [hunk 3]
```diff
@@ -55,8 +55,7 @@ export const file: GetUriProtocol<FileOptions> = async (
 
 		// `fs.ReadStream` takes care of calling `fs.close()` on the
 		// fd after it's done reading
-		// @ts-expect-error `@types/node` doesn't allow `null` as file path :/
-		const rs = createReadStream(null, {
+		const rs = createReadStream(filepath, {
```
Comment (TooTallNate) [inline packages/get-uri/src/file.ts:58]: Is this change related to the PR? I'm curious if passing in the file path will open a new FD or not.

File: packages/get-uri/test/util.ts [hunk 4]
```diff
@@ -5,5 +5,5 @@ export async function toBuffer(stream: Readable): Promise<Buffer> {
 	for await (const chunk of stream) {
 		chunks.push(chunk);
 	}
-	return Buffer.concat(chunks);
+	return Buffer.concat(chunks.map((chunk) => new Uint8Array(chunk)));
```
Comment (TooTallNate) [inline packages/get-uri/test/util.ts:8]: Curious - why was this necessary?

File: packages/get-uri/src/file.ts (hunk 3 above)
Comment (alexparish) [reply packages/get-uri/src/file.ts:58]: Yes, this change was needed as unit tests were failing in Node 14: https://github.com/TooTallNate/proxy-agents/actions/runs/11353091855/job/31595561091

You'd have thought the `@ts-expect-error` comment would have silenced that TypeError.  

Providing the filepath as the first argument of createReadStream made the unit tests pass and meant I could remove the `@ts-expect-error` comment.

Passing the filepath to createReadStream when you already have a file descriptor explicitly set in the options does not open a new file descriptor. It simply uses the one provided:

https://nodejs.org/docs/latest-v20.x/api/fs.html#:~:text=If%20fd%20is%20specified%2C%20ReadStream%20will%20ignore%20the%20path%20argument%20and%20will%20use%20the%20specified%20file%20descriptor.

File: packages/get-uri/test/util.ts (hunk 4 above)
Comment (alexparish) [reply packages/get-uri/test/util.ts]: I had a TypeError locally, but I believe I'm using a different version of TypeScript so have removed this change.

File: packages/get-uri/src/file.ts [hunk 5]
```diff
@@ -42,11 +41,12 @@ export const file: GetUriProtocol<FileOptions> = async (
...
-		const fd = await open(filepath, flags, mode);
+		const fdHandle = await fsPromises.open(filepath, flags, mode);
+		// extract the numeric file descriptor
+		const fd = fdHandle.fd;
 
-		// Now `fstat()` to check the `mtime` and store the stat object for
-		// the cache.
-		const stat = await fstat(fd);
+		// store the stat object for the cache.
+		const stat = await fdHandle.stat();
```
Comment (TooTallNate) [inline packages/get-uri/src/file.ts:49]: Is this meant to be `fstat()`? Or why was this changed?

**Comment by TooTallNate** [comment]
Thanks!
